### DISCUSSION TITLE
Charging binary sensors

### DISCUSSION THREAD

**Discussion by nathanmarlor** [nathanmarlor/foxess_modbus#69] [General]
![image](https://user-images.githubusercontent.com/3436911/226876768-af31adf9-6811-4f79-9a8c-4060a0ab1a68.png)

Currently set as `BinarySensorDeviceClass.BATTERY_CHARGING` which resolves to 'Charging' / 'Not charging'; could this be confusing for a user? Should we revert to 'On' / 'Off'? Thoughts please! :)

**Comment by canton7** [comment]
Weird, on my setup that comes up as "on" / "off". Maybe there's a HA version difference in there somewhere. Agreed that "on" / "off" is better.

One other thing to bear in mind... I was looking at implementing a service to change the charge windows. I can make a parameter which is the inverter device to change, and I can [filter that to inverters which supply an entity of type `battery_charging`](https://www.home-assistant.io/docs/blueprint/selectors/#device-selector). So only inverters which are a compatible model and which are connected over RS485 show up in the selector in the UI.

If we move away from `battery_charging` here, it would be nice to keep some way of letting the service UI filter to just inverters which supply charge windows.

**Reply by nathanmarlor** [reply]
You're right, it's the difference between 2023.3 -> 2023.4, with .4 changing to the preferred behaviour of on/off (for now).

**Reply by canton7** [reply]
Huh, that seems like a weird thing to regress on. Maybe it's accidental. I can't see any issues / PRs which mention battery_charging

**Reply by ryaaanmorgan** [reply]
Just come across this and thought it was weird that it said "charging" - I was already on version 2023.4 when moving over so thought nothing else of it!

**Reply by screenagerbe** [reply]
I'm on 2023.4.6 and I have charging/not charging in my dashboards.
To be honest I can't remember it was ever on/off

**Reply by canton7** [reply]
Maybe we should use the "running" or "power" device class? https://www.home-assistant.io/integrations/binary_sensor/#device-class

**Comment by canton7** [comment]
Fixed in #218
